toolkit/tools/pkg/imagecustomizerlib/customizeos.go

```diff
@@ -30,47 +32,70 @@ func doOsCustomizations(ctx context.Context, rc *ResolvedConfig, imageConnection
 		return err
 	}
 
-	err = addRemoveAndUpdatePackages(ctx, rc.BuildDirAbs, rc.BaseConfigPath, rc.Config.OS, imageChroot, nil,
-		rc.Options.RpmsSources, rc.Options.UseBaseImageRpmRepos, distroHandler, rc.PackageSnapshotTime)
-	if err != nil {
-		return err
-	}
+	for _, configWithBase := range rc.ConfigChain {
+		snapshotTime := configWithBase.Config.OS.Packages.SnapshotTime
+		if rc.Options.PackageSnapshotTime != "" {
+			snapshotTime = rc.Options.PackageSnapshotTime
+		}
 
-	err = UpdateHostname(ctx, rc.Config.OS.Hostname, imageChroot)
-	if err != nil {
-		return err
+		if snapshotTime != "" {
+			if !slices.Contains(configWithBase.Config.PreviewFeatures, imagecustomizerapi.PreviewFeaturePackageSnapshotTime) {
```

Comment:
This is already checked in `ImageCustomizerOptions.verifyPreviewFeatures` and `Config.IsValid`. So, I don't think you need to duplicate the check here.